### ISSUE TITLE
EuroScope Crash on Close (Linux with Wine)

### ISSUE THREAD

**Issue by sw104** [pierr3/vSMR#23]
If the vSMR plugin is loaded in EuroScope when I close EuroScope and get past the saving dialogs it crashes throwing an unhandled exception (page fault on read access) - this does not seem to happen when using any vSMR plugin from 20th of March or before, but when using any beta released on the 27th of March or later this problem occurs.

I am running an unusual setup (Linux Mint 17.3, with the 3.19.0-32 Linux kernel while running EuroScope using Wine), so I fully understand if you don't want to waste time trying to fix what is a minority issue.

Here is the crash report:

```
Unhandled exception: page fault on read access to 0x00000008 in 32-bit code (0x03e531b7).
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:006b
 EIP:03e531b7 ESP:0033ea00 EBP:0033ea0c EFLAGS:00010206(  R- --  I   - -P- )
 EAX:0409c724 EBX:00000000 ECX:0409c724 EDX:00000000
 ESI:0033ea7c EDI:00000001
Stack dump:
0x0033ea00:  0033ea20 00000016 00000000 0033ea4c
0x0033ea10:  03e532bd 0033ea20 9e8d04c1 0033eb0c
0x0033ea20:  0033ea08 0033eae8 0033ea60 00000010
0x0033ea30:  03e56338 06720100 00000000 00000010
0x0033ea40:  0033eae8 040151a0 00000000 0033eaa4
0x0033ea50:  03e5336e 0033ea7c 00000016 0409c724
Backtrace:
=>0 0x03e531b7 in vsmr (+0x31b7) (0x0033ea0c)
  1 0x03e532bd in vsmr (+0x32bc) (0x0033ea4c)
  2 0x03e5336e in vsmr (+0x336d) (0x0033eaa4)
  3 0x03fd52fe in vsmr (+0x1852fd) (0x0033eaf4)
  4 0x03fd536d in vsmr (+0x18536c) (0x0033eb20)
  5 0x03e6e8f0 in vsmr (+0x1e8ef) (0x0033eb5c)
  6 0x0049b141 in euroscope (+0x9b140) (0x0033eb88)
  7 0x0049b779 in euroscope (+0x9b778) (0x0033eba0)
  8 0x0047df62 in euroscope (+0x7df61) (0x0033f050)
  9 0x0044a5fa in euroscope (+0x4a5f9) (0x0033f080)
  10 0x0044dc58 in euroscope (+0x4dc57) (0x0033f134)
  11 0x00367d2a in cachexdll (+0x27d29) (0x0033f15c)
  12 0x0036ad08 in cachexdll (+0x2ad07) (0x0033f1d0)
  13 0x78d9f7ea in mfc100 (+0x23f7e9) (0x0033f298)
  14 0x78d9f312 in mfc100 (+0x23f311) (0x0033f2b8)
  15 0x78d9d6ff in mfc100 (+0x23d6fe) (0x0033f330)
  16 0x78d9d98b in mfc100 (+0x23d98a) (0x0033f354)
  17 0x78c959bd in mfc100 (+0x1359bc) (0x0033f398)
  18 0x7ecce4ea WINPROC_wrapper+0x19() in user32 (0x0033f3c8)
  19 0x7eccec26 in user32 (+0x9ec25) (0x0033f418)
  20 0x7eccf169 in user32 (+0x9f168) (0x0033f8e8)
  21 0x7ecd1426 in user32 (+0xa1425) (0x0033f938)
  22 0x7ec92425 DispatchMessageW+0xb4() in user32 (0x0033fa38)
  23 0x7ec607b1 IsDialogMessageW+0x130() in user32 (0x0033fb98)
  24 0x7ec920d3 IsDialogMessageA+0x72() in user32 (0x0033fbf8)
  25 0x78db4459 in mfc100 (+0x254458) (0x0033fc14)
  26 0x78da24f7 in mfc100 (+0x2424f6) (0x0033fc20)
  27 0x78d3f36e in mfc100 (+0x1df36d) (0x0033fc58)
  28 0x0045bf74 in euroscope (+0x5bf73) (0x0033fd44)
  29 0x78da06a4 in mfc100 (+0x2406a3) (0x0033fd58)
  30 0x78d8bf44 in mfc100 (+0x22bf43) (0x0033fd70)
  31 0x78d8c749 in mfc100 (+0x22c748) (0x0033fd7c)
  32 0x78d8bf94 in mfc100 (+0x22bf93) (0x0033fd88)
  33 0x78d8be79 in mfc100 (+0x22be78) (0x0033fdb8)
  34 0x78db1ce4 in mfc100 (+0x251ce3) (0x0033fdcc)
  35 0x0052de3b in euroscope (+0x12de3a) (0x0033fe60)
  36 0x7b85e5cc call_process_entry+0xb() in kernel32 (0x0033fe78)
  37 0x7b85f653 in kernel32 (+0x4f652) (0x0033feb8)
  38 0x7bc799b0 call_thread_func_wrapper+0xb() in ntdll (0x0033fed8)
  39 0x7bc7c93d call_thread_func+0x7c() in ntdll (0x0033ffa8)
  40 0x7bc7998e RtlRaiseException+0x21() in ntdll (0x0033ffc8)
  41 0x7bc4e8fe call_dll_entry_point+0x7ed() in ntdll (0x0033ffe8)
  42 0xf758b50d wine_call_on_stack+0x1c() in libwine.so.1 (0x00000000)
  43 0xf758b5cb wine_switch_to_stack+0x2a() in libwine.so.1 (0xff8a40f8)
  44 0x7bc541e2 LdrInitializeThunk+0x3a1() in ntdll (0xff8a4158)
  45 0x7b865bdd __wine_kernel_init+0xa0c() in kernel32 (0xff8a5278)
  46 0x7bc547a3 __wine_process_init+0x192() in ntdll (0xff8a5308)
  47 0xf7588c70 wine_init+0x30f() in libwine.so.1 (0xff8a5368)
  48 0x7bf00fdc main+0xfb() in <wine-loader> (0xff8a57b8)
  49 0xf73a9af3 __libc_start_main+0xf2() in libc.so.6 (0x00000000)
0x03e531b7: call    *0x8(%edx)
Modules:
Module  Address         Debug info  Name (146 modules)
PE    340000-  39a000   Export          cachexdll
PE    3a0000-  3b3000   Deferred        euroscopeplugindll
PE    400000-  7f3000   Export          euroscope
PE    f20000-  f68000   Deferred        directsound_win32
PE   1bc0000- 1bee000   Deferred        vca 3.2
PE   3e50000- 40e8000   Export          vsmr
PE  10000000-102c6000   Deferred        ts3client_win32
PE  78050000-780b9000   Deferred        msvcp100
PE  78aa0000-78b5f000   Deferred        msvcr100
PE  78b60000-78f98000   Export          mfc100
ELF 7b800000-7ba5b000   Dwarf           kernel32<elf>
  \-PE  7b810000-7ba5b000   \               kernel32
ELF 7bc00000-7bcdb000   Dwarf           ntdll<elf>
  \-PE  7bc10000-7bcdb000   \               ntdll
ELF 7bf00000-7bf04000   Dwarf           <wine-loader>
ELF 7c3a1000-7c3be000   Deferred        libgcc_s.so.1
ELF 7c3be000-7c400000   Deferred        rsaenh<elf>
  \-PE  7c3c0000-7c400000   \               rsaenh
ELF 7c404000-7c4d3000   Deferred        crypt32<elf>
  \-PE  7c410000-7c4d3000   \               crypt32
ELF 7c4d3000-7c500000   Deferred        netapi32<elf>
  \-PE  7c4e0000-7c500000   \               netapi32
ELF 7c725000-7c758000   Deferred        secur32<elf>
  \-PE  7c730000-7c758000   \               secur32
ELF 7c758000-7c800000   Deferred        msvcrt<elf>
  \-PE  7c770000-7c800000   \               msvcrt
ELF 7cb4d000-7cbc9000   Deferred        wininet<elf>
  \-PE  7cb60000-7cbc9000   \               wininet
ELF 7cc94000-7cca9000   Deferred        schannel<elf>
  \-PE  7cca0000-7cca9000   \               schannel
ELF 7ce2a000-7ce52000   Deferred        mpr<elf>
  \-PE  7ce30000-7ce52000   \               mpr
ELF 7ce52000-7ce74000   Deferred        oleacc<elf>
  \-PE  7ce60000-7ce74000   \               oleacc
ELF 7ce74000-7cefe000   Deferred        gdiplus<elf>
  \-PE  7ce80000-7cefe000   \               gdiplus
ELF 7cefe000-7cf07000   Deferred        libogg.so.0
ELF 7cf07000-7cf33000   Deferred        libvorbis.so.0
ELF 7cf33000-7d0ab000   Deferred        libvorbisenc.so.2
ELF 7d0ab000-7d0df000   Deferred        libflac.so.8
ELF 7d0df000-7d151000   Deferred        libsndfile.so.1
ELF 7d151000-7d15b000   Deferred        libwrap.so.0
ELF 7d15b000-7d1ca000   Deferred        libpulsecommon-4.0.so
ELF 7d1ca000-7d300000   Deferred        oleaut32<elf>
  \-PE  7d1e0000-7d300000   \               oleaut32
ELF 7d400000-7d407000   Deferred        libasyncns.so.0
ELF 7d407000-7d412000   Deferred        libjson-c.so.2
ELF 7d412000-7d461000   Deferred        libpulse.so.0
ELF 7d461000-7d489000   Deferred        winepulse<elf>
  \-PE  7d470000-7d489000   \               winepulse
ELF 7d489000-7d4ab000   Deferred        mmdevapi<elf>
  \-PE  7d490000-7d4ab000   \               mmdevapi
ELF 7d4ab000-7d4b4000   Deferred        librt.so.1
ELF 7d4b4000-7d4bb000   Deferred        libffi.so.6
ELF 7d4bb000-7d4d3000   Deferred        libresolv.so.2
ELF 7d4d3000-7d51e000   Deferred        libdbus-1.so.3
ELF 7d51e000-7d55a000   Deferred        libp11-kit.so.0
ELF 7d55a000-7d56e000   Deferred        libtasn1.so.6
ELF 7d56e000-7d5f5000   Deferred        libgcrypt.so.11
ELF 7d5f5000-7d601000   Deferred        libkrb5support.so.0
ELF 7d601000-7d631000   Deferred        libk5crypto.so.3
ELF 7d631000-7d6ef000   Deferred        libkrb5.so.3
ELF 7d6ef000-7d701000   Deferred        libavahi-client.so.3
ELF 7d701000-7d70f000   Deferred        libavahi-common.so.3
ELF 7d70f000-7d7d5000   Deferred        libgnutls.so.26
ELF 7d7d5000-7d81b000   Deferred        libgssapi_krb5.so.2
ELF 7d81b000-7d888000   Deferred        libcups.so.2
ELF 7d888000-7d8c8000   Deferred        winspool<elf>
  \-PE  7d890000-7d8c8000   \               winspool
ELF 7d8c8000-7d9b3000   Deferred        comdlg32<elf>
  \-PE  7d8d0000-7d9b3000   \               comdlg32
ELF 7de15000-7de1a000   Deferred        libgpg-error.so.0
ELF 7de1a000-7de1f000   Deferred        libcom_err.so.2
ELF 7de31000-7de38000   Deferred        libnss_dns.so.2
ELF 7de40000-7de81000   Deferred        usp10<elf>
  \-PE  7de50000-7de81000   \               usp10
ELF 7de81000-7dea6000   Deferred        imm32<elf>
  \-PE  7de90000-7dea6000   \               imm32
ELF 7df0f000-7df46000   Deferred        uxtheme<elf>
  \-PE  7df20000-7df46000   \               uxtheme
ELF 7df46000-7df4c000   Deferred        libxfixes.so.3
ELF 7df4c000-7df57000   Deferred        libxcursor.so.1
ELF 7df57000-7df67000   Deferred        libxi.so.6
ELF 7df67000-7df6b000   Deferred        libxcomposite.so.1
ELF 7df6b000-7df76000   Deferred        libxrandr.so.2
ELF 7df76000-7df81000   Deferred        libxrender.so.1
ELF 7df81000-7df87000   Deferred        libxxf86vm.so.1
ELF 7df87000-7df8b000   Deferred        libxinerama.so.1
ELF 7df8b000-7df92000   Deferred        libxdmcp.so.6
ELF 7df92000-7df96000   Deferred        libxau.so.6
ELF 7df96000-7dfb8000   Deferred        libxcb.so.1
ELF 7dfb8000-7e0ec000   Deferred        libx11.so.6
ELF 7e0ec000-7e0ff000   Deferred        libxext.so.6
ELF 7e0ff000-7e103000   Deferred        libkeyutils.so.1
ELF 7e108000-7e11e000   Deferred        dwmapi<elf>
  \-PE  7e110000-7e11e000   \               dwmapi
ELF 7e120000-7e1b2000   Deferred        winex11<elf>
  \-PE  7e130000-7e1b2000   \               winex11
ELF 7e1f9000-7e222000   Deferred        libexpat.so.1
ELF 7e222000-7e25d000   Deferred        libfontconfig.so.1
ELF 7e25d000-7e285000   Deferred        libpng12.so.0
ELF 7e285000-7e29f000   Deferred        libz.so.1
ELF 7e29f000-7e33f000   Deferred        libfreetype.so.6
ELF 7e360000-7e3a9000   Deferred        dsound<elf>
  \-PE  7e370000-7e3a9000   \               dsound
ELF 7e3a9000-7e5dc000   Deferred        shell32<elf>
  \-PE  7e3c0000-7e5dc000   \               shell32
ELF 7e5dc000-7e602000   Deferred        iphlpapi<elf>
  \-PE  7e5e0000-7e602000   \               iphlpapi
ELF 7e602000-7e61e000   Deferred        wsock32<elf>
  \-PE  7e610000-7e61e000   \               wsock32
ELF 7e61e000-7e632000   Deferred        msimg32<elf>
  \-PE  7e620000-7e632000   \               msimg32
ELF 7e632000-7e739000   Deferred        comctl32<elf>
  \-PE  7e640000-7e739000   \               comctl32
ELF 7e739000-7e7b3000   Deferred        shlwapi<elf>
  \-PE  7e750000-7e7b3000   \               shlwapi
ELF 7e7b3000-7e7de000   Deferred        msacm32<elf>
  \-PE  7e7c0000-7e7de000   \               msacm32
ELF 7e7de000-7e85f000   Deferred        rpcrt4<elf>
  \-PE  7e7f0000-7e85f000   \               rpcrt4
ELF 7e85f000-7e99b000   Deferred        ole32<elf>
  \-PE  7e880000-7e99b000   \               ole32
ELF 7e99b000-7ea55000   Deferred        winmm<elf>
  \-PE  7e9a0000-7ea55000   \               winmm
ELF 7ea55000-7ea8b000   Deferred        ws2_32<elf>
  \-PE  7ea60000-7ea8b000   \               ws2_32
ELF 7ea8b000-7eafd000   Deferred        advapi32<elf>
  \-PE  7eaa0000-7eafd000   \               advapi32
ELF 7eafd000-7ec1a000   Deferred        gdi32<elf>
  \-PE  7eb10000-7ec1a000   \               gdi32
ELF 7ec1a000-7ed74000   Dwarf           user32<elf>
  \-PE  7ec30000-7ed74000   \               user32
ELF 7ed74000-7ed80000   Deferred        libnss_files.so.2
ELF 7ed80000-7ed99000   Deferred        libnsl.so.1
ELF 7ef99000-7efdf000   Deferred        libm.so.6
ELF 7efe6000-7f000000   Deferred        version<elf>
  \-PE  7eff0000-7f000000   \               version
ELF f7383000-f738f000   Deferred        libnss_nis.so.2
ELF f7390000-f753f000   Dwarf           libc.so.6
ELF f753f000-f7544000   Deferred        libdl.so.2
ELF f7545000-f7561000   Deferred        libpthread.so.0
ELF f7567000-f7570000   Deferred        libnss_compat.so.2
ELF f7582000-f7737000   Dwarf           libwine.so.1
ELF f7739000-f775b000   Deferred        ld-linux.so.2
ELF f775d000-f775e000   Deferred        [vdso].so
Threads:
process  tid      prio (all id:s are in hex)
0000000e services.exe
    0000001f    0
    0000001e    0
    00000014    0
    00000010    0
    0000000f    0
00000012 winedevice.exe
    0000001d    0
    0000001a    0
    00000017    0
    00000013    0
0000001b plugplay.exe
    00000021    0
    00000020    0
    0000001c    0
00000022 explorer.exe
    00000024    0
    00000023    0
00000025 (D) C:\Program Files (x86)\EuroScope\EuroScope.exe
    0000003a    0
    00000038    0
    00000037    0
    00000034    0
    00000033    0
    00000032    0
    00000031    0
    00000030    0
    0000002f    0
    0000002e    0
    0000002d    0
    0000002c   15
    0000002b    1
    0000002a   15
    00000029    0
    00000028    0
    00000027    0
    00000026    0 <==
System information:
    Wine build: wine-1.6.2
    Platform: i386 (WOW64)
    Host system: Linux
    Host version: 3.19.0-32-generic
```

**Comment by pierr3** [comment]
I'll try it out on my Ubuntu machine, try setting wine to use Win 7 or 10, I think this might be a weird memory management issue.

**Comment by sw104** [comment]
I have just tried it setting wine to use Windows 7 and also separately Windows 10 with the same results. I also tried using the latest Wine version (Wine Staging 1.9.14) and instead EuroScope now hangs on exit instead of throwing an exception (in the terminal window I get `err:ntdll:RtlpWaitForCriticalSection section 0x7bceb920 "loader.c: loader_section" wait timed out in thread 0030, blocked by 0009, retrying (60 sec)` every 60 seconds).
